Software/controller/controller.py

```diff
@@ -82,90 +82,96 @@ def on_message(client, userdata, msg):
-                            break
-                        if placeInList == self.model.list_of_places[-1]:
-                            # Found no vacant place - put number in queue
-                            print("New number " + str(new_number))
-                            self.model.list_of_ticket_numbers.append(new_number)
-                            self.update_queue()
+                    handleMessageFromController(json_loaded)
             except Exception as e:
                 print(str(e))
                 print("Something went wrong on mqtt reception")
+            self.update_queue(self.model.list_of_processing_times, self.model.list_of_labels_to_display_in_queue, self.model.list_of_ticket_numbers)
+
+        def handleMessageFromController(json_loaded):
+            new_number = json_loaded["new_number"]
+            print("Received new ticket number from controller: " +
+                  str(new_number))
+            if new_number < 0:
+                    raise Exception('New ticket number must be positive!')
+            # Check if new number is already in use
+            for placeInList in self.model.list_of_places:
+                if placeInList.ticket_number == new_number:
+                    print("New number " + str(new_number) +
+                          " already registered")
+                    return
+            if new_number in self.model.list_of_ticket_numbers:
+                print("New number " + str(new_number) +
+                      "already in ticket number list")
+                return
+            for count, placeInList in enumerate(self.model.list_of_places):
+                # Search an empty place
+                if placeInList.state == PlaceState.FREE:
+                    # Found one - register ticket number to place
+                    print("Register ticket: " +
+                          str(new_number) + " to free place " + str(count + 1))
+
+                    placeInList.state = PlaceState.REGISTERED
+                    placeInList.ticket_number = new_number
+                    self.model.list_of_labels_to_display_place_number[count].config(
+                        bg="green")
+                    self.model.list_of_labels_to_display_ticket_number[count].config(
+                        text=placeInList.ticket_number)
+                    break
+                if placeInList == self.model.list_of_places[-1]:
+                    # Found no vacant place - put number in queue
+                    print("New number " + str(new_number))
+                    self.model.list_of_ticket_numbers.append(new_number)
+
+        def handleMessageFromPlace(json_loaded):
+            if json_loaded["place_occupied"] == True:
+                # Place was taken by the owner of the ticket_number
+                # Place number in MQTT-Message starts with 1 and must be decremented
+                place_number = json_loaded["place_number"] - 1
```

Comment:
Since you are already moving code around and do lots of whitespace changes - what about splitting this function even more?

I'm thinking of something like `def occupyPlace(self, place_number):` and `def releasePlace(self, place_number):`

Reply:
Great idea!

Reply:
added  functions occupyPlace(self, place_number) and reservePlaceForTicketNumber(self, place_number, ticket_number)